externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -62,7 +63,15 @@ class ExecuteStatement(
       Thread.currentThread().setContextClassLoader(spark.sharedState.jarClassLoader)
       spark.sparkContext.setJobGroup(statementId, statement)
       result = spark.sql(statement)
-      iter = result.collect().toList.iterator
+      val castCols = result.schema.map { field =>
+        field.dataType match {
+          case BooleanType | ByteType | ShortType | IntegerType | LongType |
```

Comment:
`ByteType | ShortType | IntegerType | LongType | FloatType | DoubleType` to  _: NumericType for simplicity

Reply:
also add StringType to avoid casting

Reply:
`DecimalType` also is `NumericType` but not present here.

Reply:
I think `IntegralType | FloatType | DoubleType` is more precise since `DecimalType` is `NumericType` but should not be included.

Reply:
Oops! We can't use `IntegralType` because it is `private[sql]`
```
private[sql] abstract class IntegralType extends NumericType
```

Reply:
`IntegralType` isn't public, Let's leave it

Reply:
updated